modules/reindex/src/main/java/org/elasticsearch/index/reindex/AbstractBulkByQueryRestHandler.java

```diff
@@ -83,6 +83,9 @@ protected void parseInternalRequest(Request internal, RestRequest restRequest,
      */
     private XContentParser extractRequestSpecificFields(RestRequest restRequest,
                                                         Map<String, Consumer<Object>> bodyConsumers) throws IOException {
+        if(restRequest.hasContentOrSourceParam() == false) {
```

Comment:
Jumped the gun on cleaning up this class: turns out that `update_by_query` allows an empty body so we need to make sure this method does not throw an exception is the request body or source param is missing.